tests/integration/test_charm.py

```diff
@@ -361,3 +365,107 @@ def generate_s3_config(s3_url: str) -> Dict:
     s3_config["ip_address"] = s3_ip_address
     s3_config["endpoint"] = s3_endpoint
     return s3_config
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.requires_secrets
+@pytest.mark.usefixtures("setup_saml_config")
+async def test_saml_login(  # pylint: disable=too-many-locals
+    ops_test: OpsTest,
+    app: Application,
+    pytestconfig: pytest.Config,
+    requests_timeout: int,
+    run_action,
+):
+    """
+    arrange: after discourse charm has been deployed, with all required relation established.
+    act: add an admin user and enable force-https mode.
+    assert: user can login discourse using SAML Authentication.
+    """
+    assert ops_test.model
```

Comment:
Would it be worth making this a fixture so the `assert ops_test.model` is only required once in that fixture?

Reply:
I don't think mypy can get context across fixtures.

Reply:
What I was thinking was that the fixture would be something like this:

```python
@pytest.fixture
def model(ops_test):
    assert ops_test.model
    yield model
```

And then the test would be:

```python
def test_something(model: Model):
    ...
```

Reply:
So the fixture guarantees that the model is not `None` and then the type hint for the fixture can just be the model

Reply:
Updated, thanks!